openapi/openapi_v2_backend_configuration_test.go

```diff
@@ -124,6 +124,302 @@ func TestGetHost(t *testing.T) {
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.${region}.some-backend.com",
+					extTfProviderRegions:         "rst1,dub1",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When isMultiRegion() method is called", func() {
+			isMultiRegion, host, regions, err := specV2BackendConfiguration.isMultiRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("Then it should be multi region", func() {
+				So(isMultiRegion, ShouldBeTrue)
+			})
+			Convey("Then host should be the parametrised host", func() {
+				So(host, ShouldEqual, "www.${region}.some-backend.com")
+			})
+			Convey("Then regions should contain the right regions", func() {
+				So(regions, ShouldContain, "rst1")
+				So(regions, ShouldContain, "dub1")
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is NOT multi-region", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When isMultiRegion() method is called", func() {
+			isMultiRegion, _, _, err := specV2BackendConfiguration.isMultiRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("Then it should be multi region", func() {
+				So(isMultiRegion, ShouldBeFalse)
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is multi-region but the x-terraform-provider-multiregion-fqdn does not have a parametrised value", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.some-backend.com",
+					extTfProviderRegions:         "rst1",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When isMultiRegion() method is called", func() {
+			isMultiRegion, _, _, err := specV2BackendConfiguration.isMultiRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "'x-terraform-provider-multiregion-fqdn' extension value provided not matching multiregion host format")
+			})
+			Convey("Then it should be multi region", func() {
+				So(isMultiRegion, ShouldBeFalse)
+			})
+		})
+	})
+
+	Convey("Given a specV2BackendConfiguration that is multi-region but the x-terraform-provider-multiregion-fqdn does not have a parametrised value", t, func() {
```

Comment:
description should cover provider regions being empty

Reply:
done